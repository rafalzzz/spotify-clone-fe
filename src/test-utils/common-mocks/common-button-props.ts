export const commonButtonProps = {
  type: 'default',
  htmlType: 'button',
  onClick: jest.fn(),
};
