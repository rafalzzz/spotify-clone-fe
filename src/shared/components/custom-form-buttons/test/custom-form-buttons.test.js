import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';

import { CustomFormButtons } from '..';

import { commonButtonProps } from '@/test-utils/common-mocks/common-button-props';

import '@testing-library/jest-dom';

const BUTTON_TEXT_MOCKS = ['Test Button', 'Test Button2'];

const BUTTONS_DATA_MOCK = [
  {
    ...commonButtonProps,
    text: BUTTON_TEXT_MOCKS[0],
    key: 'Test1',
  },
  {
    ...commonButtonProps,
    text: BUTTON_TEXT_MOCKS[1],
    key: 'Test2',
  },
];

const renderCustomFormButtons = () => {
  return render(<CustomFormButtons formButtons={BUTTONS_DATA_MOCK} />);
};

describe('CustomFormButtons', () => {
  it('render component without error', () => {
    const screen = renderCustomFormButtons();
    expect(screen).toMatchSnapshot();
  });

  it('renders the correct text', () => {
    const { findByRole } = renderCustomFormButtons();

    BUTTON_TEXT_MOCKS.forEach(async (buttonText) => {
      const button = await findByRole('button', { name: buttonText });
      expect(button).toBeInTheDocument();
      expect(buttonElement).toHaveTextContent(buttonText);
    });
  });
});
