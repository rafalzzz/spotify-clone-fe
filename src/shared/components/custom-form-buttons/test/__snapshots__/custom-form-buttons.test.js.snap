// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomFormButtons render component without error 1`] = `undefined`;
